twisted/test/proto_helpers.py

```diff
@@ -695,3 +702,86 @@ def connectUNIX(self, address, factory, timeout=30, checkPID=0):
         Fake L{IReactorUNIX.connectUNIX}, that raises L{_connectException}.
         """
         raise self._connectException
+
+
+
+@implementer(ILogFilterPredicate)
+class NamespaceFilterPredicate(object):
+    """
+    L{ILogFilterPredicate} that filters out events not matching configured
+    namespace.
+    """
+
+    def __init__(self, namespace):
+        self.namespace = namespace
+
+
+    def __call__(self, event):
+        namespace = event.get("log_namespace", None)
+
+        if namespace != self.namespace:
+            return PredicateResult.no
+
+        return PredicateResult.maybe
+
+
+
+class LogCapture(object):
+    """
+    A context manager that captures log events and makes them available in
+    plain text format or as raw events list.
+
+    Can be used like this:
+
+    with LogCapture() as lc:
+        [... code that logs events ...]
+
+        events = lc.events
+        messages = lc.messages()
+        self.assertIn('logline text', messages[0])
+        self.assertNotIn('Unable to format event', messages[0])
+        selc.assertEqual(events['log_level'], Loglevel.debug)
+
+    """
```

Comment:
This docstring is missing an `@ivar` for `namespace` and `_events`.  Perhaps `_namespace` could be public?